magefiles/magefile.go

```diff
@@ -627,41 +643,102 @@ func SetupSourceBuild() {
 	}
 	dockerPipelineObject := tektonObj.(*tektonapi.Pipeline)
 
-	// Update build-source-image param value to true
-	for i := range dockerPipelineObject.PipelineSpec().Params {
-		if dockerPipelineObject.PipelineSpec().Params[i].Name == "build-source-image" {
-			//fmt.Printf("Current build-source-image param default value: %q\n", dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal)
-			dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal = "true"
+	if source_build {
+		// Update build-source-image param value to true
+		for i := range dockerPipelineObject.PipelineSpec().Params {
+			if dockerPipelineObject.PipelineSpec().Params[i].Name == "build-source-image" {
+				//fmt.Printf("Current build-source-image param default value: %q\n", dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal)
+				dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal = "true"
+			}
+		}
+		// Update the source-build task image reference to SOURCE_BUILD_IMAGE
+		var currentSourceTaskBundle string
+		for i := range dockerPipelineObject.PipelineSpec().Tasks {
+			t := &dockerPipelineObject.PipelineSpec().Tasks[i]
+			params := t.TaskRef.Params
+			var lastBundle *tektonapi.Param
+			sourceTask := false
+			for i, param := range params {
+				if param.Name == "bundle" {
+					lastBundle = &t.TaskRef.Params[i]
+				} else if param.Name == "name" && param.Value.StringVal == "source-build" {
+					sourceTask = true
```

Comment:
We can `break` at this point

Reply:
Good catch. Added now.

Reply:
Params can be in any order. If we use `break` we might have `lastBundle` as `nil`  after this loop. For this example below:
```
    taskRef:
      params:
      - name: name
        value: source-build
      - name: bundle
        value: quay.io/konflux-ci/tekton-catalog/task-source-build:0.2@sha256:e8c321b8a67e421a9c3975fd9a938ca4e838976064e14c7c0eb4e1f261900b1c
      - name: kind
        value: task
      resolver: bundles

```

Reply:
Yes, you are right